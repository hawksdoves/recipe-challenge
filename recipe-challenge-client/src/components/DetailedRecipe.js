import React from 'react';
import { Link } from 'react-router-dom';
import Recipe from './Recipe';
import getRecipe from '../services/getRecipe';

class DetailedRecipe extends React.Component {

    constructor(props) {
        super(props)
        this.state = { recipe: undefined };
    }

    componentDidMount() {
        const { id } = this.props.match.params;
        getRecipe(id)
            .then(resp => this.setState({ recipe: resp.data }))
    }

    render() {
        return (
            <div>
                <Link to='/recipes'>Return to list</Link >
                <Recipe recipe={this.state.recipe} />
            </div>
        )
    }
    
};

export default DetailedRecipe;