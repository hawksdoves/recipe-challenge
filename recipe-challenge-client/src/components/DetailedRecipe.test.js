import React from 'react';
import { shallow, mount } from 'enzyme';

import DetailedRecipe from './DetailedRecipe';

describe('DetailedRecipe', () => {
    it('should render the recipe correctly', () =>{
        const wrapper = shallow(<DetailedRecipe />);

        expect(wrapper).toMatchSnapshot()
    })

    // it('calls componentDidMount() lifecycle method', () => {
    //     const mockParamsId = { params: { id: '1' } };
    //     const componentDidMountSpy = jest.spyOn(DetailedRecipe.prototype, 'componentDidMount');
    //     const wrapper = mount(<DetailedRecipe />);

    //     expect(componentDidMountSpy).toHaveBeenCalled();

    //     componentDidMountSpy.restore();
    // });
});
