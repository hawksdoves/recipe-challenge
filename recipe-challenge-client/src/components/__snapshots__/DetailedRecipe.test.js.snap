// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedRecipe should render the recipe correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <Link
        replace={false}
        to="/recipes"
    >
        Return to list
    </Link>
    <Recipe
        recipe={undefined}
    />
</div>,
  "nodes": Array [
    <div>
      <Link
            replace={false}
            to="/recipes"
      >
            Return to list
      </Link>
      <Recipe
            recipe={undefined}
      />
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <DetailedRecipe />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": DetailedRecipe {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {
          "recipe": undefined,
        },
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <Link
                    replace={false}
                    to="/recipes"
          >
                    Return to list
          </Link>
          <Recipe
                    recipe={undefined}
          />
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <Link
                    replace={false}
                    to="/recipes"
          >
                    Return to list
          </Link>
          <Recipe
                    recipe={undefined}
          />
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <DetailedRecipe />,
}
`;
